src/cluster_legacy.h

```diff
@@ -267,9 +267,9 @@ static_assert(offsetof(clusterMsg, data) == 2256, "unexpected field offset");
 /* Message flags better specify the packet content or are used to
  * provide some information about the node state. */
 #define CLUSTERMSG_FLAG0_PAUSED (1<<0) /* Master paused for manual failover. */
-#define CLUSTERMSG_FLAG0_FORCEACK (1<<1) /* Give ACK to AUTH_REQUEST even if
-                                            master is up. */
+#define CLUSTERMSG_FLAG0_FORCEACK (1<<1) /* Give ACK to AUTH_REQUEST even if master is up. */
 #define CLUSTERMSG_FLAG0_EXT_DATA (1<<2) /* Message contains extension data */
+#define CLUSTERMSG_FLAG0_NODE_SUPPORTS_EXTENSIONS (1<<3) /* Node supports extension parsing */
```

Comment:
Now I am not sure if we even need this flag - I think it is a dup of `CLUSTERMSG_FLAG0_EXT_DATA`.

Reply:
No, it's not the same. We need a way to indicate we are sending extensions without sending the extensions. Otherwise it'll get evaluated as if this node has zero extensions, which breaks the logic for hostnames and the TLS messages, since we assume if a node doesn't send them it removed them.

Reply:
I think I agree with you @PingXie . @madolson and I discussed a bit more on this and the problem (display nodes only on receiving complete information) we're trying to solve seems not practical without patching older version(s). Will incorporate this feedback.